llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -745,8 +745,8 @@ void NVPTXAsmPrinter::emitHeader(Module &M, raw_ostream &O,
     switch(CU->getEmissionKind()) {
     case DICompileUnit::NoDebug:
     case DICompileUnit::DebugDirectivesOnly:
-      break;
     case DICompileUnit::LineTablesOnly:
+      break;
```

Comment:
@dwblaikie Do I understand it correctly that `LineTablesOnly` only implies that debug info will be limited to line info only, but it does not say anything about the format of that info. I.e. that it could be either done via debug directives or via DWARF.

If that's the case, then this change is wrong, and the current code is correct.

Reply:
LineTablesOnly unfortunately means /almost/ line tables only - it actually means line tables, /and/ enough of .debug_info to describe inlining.

The DebugDirectivesOnly is means /really/ only the debug directives that generate only the line table (.debug_line (and in DWARFv5, .debug_line_str)).